### DISCUSSION TITLE
Slow Image Render With Huge Images -- Regression since Avalonia 10?

### DISCUSSION THREAD

**Discussion by alexhelms** [AvaloniaUI/Avalonia#13779] [General]
I'm picking up a side project that I haven't worked on much since before Avalonia 11 was released. I'm displaying 60MP images using a `WriteableBitmap` but there seems to have been some kind of regression in Avalonia. The image is wrapped in a `ZoomBorder` and panning the image around dramatically drops the FPS -- this used to be very smooth and I don't believe any of my application code has changed to cause this problem.

Using dotTrace while panning the image, I see it is spending a lot of time drawing and disposing the SkiaImage.

![image](https://github.com/AvaloniaUI/Avalonia/assets/174474/9ad23ead-ecf9-49d0-b4b4-23a3185adbf0)

I have `MaxGpuResourcesSizeBytes` set to 1GB. If I force software rendering only, the performance when panning is fine.

I'm running Windows 11 23H2.

What else can I try?

**Comment by timunie** [comment]
which Avalonia version btw? Maybe also worth to check with nightly builds btw.

**Reply by alexhelms** [reply]
Oops, this is in 11.0.5. I just tried with the latest nightly `11.1.999-cibuild0042513-beta` and I don't see any change.

**Reply by thevortexcloud** [reply]
I have not tried anything but 11.0.5 but I can also confirm zooming and panning with very large images is fairly slow. But it's not a huge blocker for me right now. "Small" images work fine. Interestingly zooming into the image makes it also work fine (although in my case I have to zoom fairly far due to how big the picture is).

**Comment by timunie** [comment]
maybe skia3 will bring a performance boost here as well. Worth to test using #12729 even if it is not yet production ready. Other than that I have no idea beside using a reduced resolution / cropped image as needed.

**Reply by alexhelms** [reply]
How could I use that PR's build? Sorry, I'm no familiar with using the nightlies.

**Reply by timunie** [reply]
Oh looks like the builds are removed for closed PRs. So you may need to check out this branch: https://github.com/adirh3/Avalonia/tree/features/skia-sharp-3.0 and add your sample to `src/samples/sandbox`
